src/main/java/com/synopsys/integration/blackduck/codelocation/upload/UploadOutput.java

```diff
@@ -17,26 +17,32 @@
 
 public class UploadOutput extends CodeLocationOutput {
     private final String response;
+    private final String scanId;
 
-    private UploadOutput(@Nullable NameVersion projectAndVersion, String codeLocationName, Result result, String response, String errorMessage, Exception exception) {
+    private UploadOutput(@Nullable NameVersion projectAndVersion, String codeLocationName, Result result, String response, String errorMessage, Exception exception, String scanId) {
         super(result, projectAndVersion, codeLocationName, 1, errorMessage, exception);
         this.response = response;
+        this.scanId = scanId;
     }
 
-    public static UploadOutput SUCCESS(@Nullable NameVersion projectAndVersion, String codeLocationName, String response) {
-        return new UploadOutput(projectAndVersion, codeLocationName, Result.SUCCESS, response, null, null);
+    public static UploadOutput SUCCESS(@Nullable NameVersion projectAndVersion, String codeLocationName, String response, String scanId) {
```

Comment:
I think it would require some digging by someone writing new upload code to know that it's OK to pass null for ScanId. Seems better to have two versions of this SUCCESS method, one that with the scanId arg, one without. Or, at a minimum, I think the arg should be marked @Nullable.

Reply:
I too prefer overloading to handling nullable arguments.